pkg/jobs/wait.go

```diff
@@ -118,6 +119,12 @@ func (r *Registry) waitForJobsToBeTerminalOrPaused(
 			query,
 		)
 		if err != nil {
+			// While polling, we could encounter retryable errors, which may
+			// not get internally retried in the connection executor. So retry,
+			// here.
+			if errors.HasInterface(err, (*pgerror.ClientVisibleRetryError)(nil)) {
```

Comment:
nice find! a few semi-related comments before we go with this:
- could you add a comment that mentions that this is safe to retry here because we use a new txn for each query?
- also i'm a bit surprised - the internal executor is supposed to retry this automatically. so it feels like something else must be going wrong if that isn't working. 
- could we avoid retry errors by making `makeWaitForJobsQuery` use a fixed timestamp?
- there should be a test that reproduces the problem reliably so we can make sure the fix works going forwards. this test would also help us understand why the internal executor did not retry.

Reply:
> also i'm a bit surprised - the internal executor is supposed to retry this automatically. so it feels like something else must be going wrong if that isn't working.

@rafiss It's because results are returned to the client, so it's no longer safe to retry internally in this case. Let me double check and confirm that one more time too